Adafruit_TLC59711.cpp

```diff
@@ -29,25 +37,37 @@ Adafruit_TLC59711::Adafruit_TLC59711(uint8_t n, uint8_t c, uint8_t d) {
   pwmbuffer = (uint16_t *)calloc(2, 12*n);
 }
 
-Adafruit_TLC59711::Adafruit_TLC59711(uint8_t n) {
+/*!
+ *  @brief  Instantiates a new Adafruit_TLC59711 class using provided SPI
+ *  @param  n
+ *          number of connected drivers
+ *  @param  *theSPI
+ *          spi object
+ */
+Adafruit_TLC59711::Adafruit_TLC59711(uint8_t n, SPIClass *theSPI) {
   numdrivers = n;
   _clk = -1;
   _dat = -1;
+  _spi = theSPI;
 
-  SPI.setBitOrder(MSBFIRST);
+  _spi->setBitOrder(MSBFIRST);
 #ifdef __arm__
-  SPI.setClockDivider(42);
+  _spi->setClockDivider(42);
 #else
-  SPI.setClockDivider(SPI_CLOCK_DIV8);
+  _spi->setClockDivider(SPI_CLOCK_DIV8);
 #endif
-  SPI.setDataMode(SPI_MODE0);
+  _spi->setDataMode(SPI_MODE0);
   BCr = BCg = BCb = 0x7F;
 
   pwmbuffer = (uint16_t *)calloc(2, 12*n);
 }
 
+/*!
+ *  @brief  Write data throught SPI at MSB
+ *  @param  d
+ *          data
+ */
 void  Adafruit_TLC59711::spiwriteMSB(uint32_t d) {
```

Comment:
odd, `d` should be uint8_t not 32, please try changing it?

Reply:
done